From 4648041ecc5e3ccc0eb476dd6e5c6cb0de580a4a Mon Sep 17 00:00:00 2001
From: Tero Kristo <t-kristo@ti.com>
Date: Thu, 17 Mar 2016 02:56:51 -0500
Subject: [PATCH 019/347] ARM: dts: keystone-k2g: Add clock support

This patch comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

Add k2g_clks node, which acts as the clock provider for the whole
system. Any clock users must ask for their clocks via this node.
Add clock nodes to all appropriate device nodes that exists.

Signed-off-by: Tero Kristo <t-kristo@ti.com>
Tested-by: Dave Gerlach <d-gerlach@ti.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
(cherry picked from commit 1647f458218897a8a67e99bd065855bcad3032c5)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/keystone-k2g.dtsi |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/arch/arm/boot/dts/keystone-k2g.dtsi b/arch/arm/boot/dts/keystone-k2g.dtsi
index 830c3dd..6c7490b 100644
--- a/arch/arm/boot/dts/keystone-k2g.dtsi
+++ b/arch/arm/boot/dts/keystone-k2g.dtsi
@@ -16,6 +16,7 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/pinctrl/keystone.h>
 #include <dt-bindings/genpd/k2g.h>
+#include <dt-bindings/clock/k2g.h>
 #include "skeleton.dtsi"
 
 / {
@@ -91,7 +92,9 @@
 			reg-io-width = <4>;
 			reg = <0x02530c00 0x100>;
 			interrupts = <GIC_SPI 164 IRQ_TYPE_EDGE_RISING>;
-			clock-frequency = <200000000>;
+			clocks = <&k2g_clks K2G_DEV_UART0 0>;
+			clock-names = "fck";
+
 			status = "disabled";
 			power-domains = <&k2g_pds K2G_DEV_UART0>;
 		};
@@ -117,6 +120,12 @@
 			reg = <0x02921c00 0x400>;
 		};
 
+		k2g_clks: k2g_clks {
+			compatible = "ti,sci-clk";
+			ti,sci = <&pmmc>;
+			#clock-cells = <2>;
+		};
+
 		k2g_pds: k2g_pds {
 			compatible = "ti,sci-pm-domains";
 			#power-domain-cells = <1>;
-- 
1.7.5.4

